internal/storage/rolebinding.go

```diff
@@ -0,0 +1,247 @@
+	tx, err := getContextTx(ctx)
+	if err != nil {
+		return types.RoleBinding{}, err
+	}
+
+	var rb types.RoleBinding
+
+	err = tx.QueryRowContext(ctx, `
+		INSERT INTO rolebindings (id, resource_id, created_by, updated_by, created_at, updated_at)
+		VALUES ($1, $2, $3, $3, $4, $4)
+		RETURNING id, resource_id, created_by, updated_by, created_at, updated_at
+		`, rbID.String(), resourceID.String(), actorID.String(), time.Now(),
+	).Scan(
+		&rb.ID,
+		&rb.ResourceID,
+		&rb.CreatedBy,
+		&rb.UpdatedBy,
+		&rb.CreatedAt,
+		&rb.UpdatedAt,
+	)
+	if err != nil {
+		return types.RoleBinding{}, fmt.Errorf("%w: %s", err, rbID.String())
+	}
+
+	return rb, nil
+}
+
+func (e *engine) UpdateRoleBinding(ctx context.Context, actorID, rbID gidx.PrefixedID) (types.RoleBinding, error) {
+	tx, err := getContextTx(ctx)
+	if err != nil {
+		return types.RoleBinding{}, err
+	}
+
+	var rb types.RoleBinding
+
+	err = tx.QueryRowContext(ctx, `
+		UPDATE rolebindings
+		SET updated_by = $1, updated_at = now()
+		WHERE id = $2
+		RETURNING id, resource_id, created_by, updated_by, created_at, updated_at
+		`,
+		actorID.String(), rbID.String(),
+	).Scan(
+		&rb.ID,
+		&rb.ResourceID,
+		&rb.CreatedBy,
+		&rb.UpdatedBy,
+		&rb.CreatedAt,
+		&rb.UpdatedAt,
+	)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return types.RoleBinding{}, fmt.Errorf("%w: %s", ErrRoleBindingNotFound, rbID.String())
+		}
+
+		return types.RoleBinding{}, fmt.Errorf("%w: %s", err, rbID.String())
+	}
+
+	return rb, nil
+}
+
+func (e *engine) DeleteRoleBinding(ctx context.Context, id gidx.PrefixedID) error {
+	tx, err := getContextTx(ctx)
+	if err != nil {
+		return err
+	}
+
+	result, err := tx.ExecContext(ctx, `
+		DELETE FROM rolebindings WHERE id = $1
+		`, id.String(),
+	)
+	if err != nil {
+		return fmt.Errorf("%w: %s", err, id.String())
+	}
+
+	rowsAffected, err := result.RowsAffected()
+	if err != nil {
+		return fmt.Errorf("%w: %s", err, id.String())
+	}
+
+	if rowsAffected == 0 {
+		return fmt.Errorf("%w: %s", ErrRoleBindingNotFound, id.String())
+	}
+
+	return nil
+}
+
+func (e *engine) LockRoleBindingForUpdate(ctx context.Context, id gidx.PrefixedID) error {
+	db, err := getContextDBQuery(ctx, e)
+	if err != nil {
+		return err
+	}
+
+	result, err := db.ExecContext(ctx, `SELECT 1 FROM rolebindings WHERE id = $1 FOR UPDATE`, id.String())
+	if err != nil {
+		return err
+	}
+
+	rowsAffected, err := result.RowsAffected()
+	if err != nil {
+		return err
+	}
+
+	if rowsAffected == 0 {
+		return ErrRoleBindingNotFound
+	}
+
+	return nil
+}
+
+// buildBatchInClauseWithIDs is a helper function that builds an IN clause for
+// a batch query with the provided prefixed IDs.
+func (e *engine) buildBatchInClauseWithIDs(ids []gidx.PrefixedID) (clause string, args []any) {
+	args = make([]any, len(ids))
+
+	for i, id := range ids {
+		fmtStr := "$%d"
+
+		if i > 0 {
+			fmtStr = ", $%d"
+		}
+
+		clause += fmt.Sprintf(fmtStr, i+1)
+		args[i] = id.String()
+	}
+
+	return clause, args
+}
```

Comment:
Do we still need this?

Reply:
it also used in `rolev2` where we need to grab a bunch of roles, might as well keep it